### ISSUE TITLE
[Bug]: KryptonTreeView can throw exception on changing palette via KryptonManager

### ISSUE THREAD

**Issue by ZXBITLES** [Krypton-Suite/Standard-Toolkit#777]
**Describe the bug**
Exception can be thrown in KryptonTreeView by method WndProc, when KryptonTreeView is visible (has expanded nodes) and palette changed.

**To Reproduce**
1. Create a form with KryptonTreeView.
2. Fill KryptonTreeView with some nodes.
3. Add KryptonManager to a form.
4. Start app.
5. Expand few nodes in KryptonTreeView.
6. Change palette via KryptonManager.
7. Exception will be thrown in KryptonTreeView.cs at 
protected override void WndProc(ref Message m) {
...
 default:
                        base.WndProc(ref m);
                        break;
...
}

**Expected behavior**
Exception shouldn't be thrown.

**Comment by Wagnerp** [comment]
@Smurf-IV Any ideas?

**Comment by ZXBITLES** [comment]
Well, I made a stupid workaround and its basically works:
default:
                        try
                        {
                            base.WndProc(ref m);
                        }
                        catch (Exception)
                        { }
                        break;
I think, the reason is that some nodes not available or not exist, so here we should write some kind if condition to check if it possible to call "base.WndProc(ref m);" or not, but I have no idea what to check...

By the way, for some reasons KryptonTreeView trying to collapse all the nodes when changing a palette, and when its done, then no exception is thrown.

**Comment by Smurf-IV** [comment]
> @Smurf-IV Any ideas?

I have not seen this in the demo code

**Comment by Smurf-IV** [comment]
> Well, I made a stupid workaround and its basically works: default: try { base.WndProc(ref m); } catch (Exception) { } break;

Where did you place this code ?

**Comment by Wagnerp** [comment]
> > Well, I made a stupid workaround and its basically works: default: try { base.WndProc(ref m); } catch (Exception) { } break;
> 
> Where did you place this code ?

Will ask OP

**Comment by ZXBITLES** [comment]
> Where did you place this code ?
In "KryptonTreeView.cs" at "protected override void WndProc(ref Message m)".
By the way, I am creating KryptonManager dynamically on starting App before showing any forms and working with it to change a palette. So, KryptonManager does not lying on form. I don't think, that this is important, but must mention this.

**Comment by ZXBITLES** [comment]
Made a test - put KryptonManager to a form. The same result.
Besides of it, I see, that KryptonTreeView collapsing all the nodes on changing a palette. Could be that this can give an idea to find a reason.

**Comment by ZXBITLES** [comment]
Ok.
I think, that I got it!
Let me describe.
I created my own control which contains KryptonTreeView.
I am handling AfterSelect event for this KryptonTreeView.
The way to solve the problem is to check if e.Action equals to TreeViewAction.Unknown and exit a method in this case.
However, I do believe, that this is a bug and hope that my explanation can help to locate it.

"Bad" code:
Private Sub _TrvItems_AfterSelect(sender As Object, e As TreeViewEventArgs) Handles _TrvItems.AfterSelect
        ...
End Sub 

"Good" code:
Private Sub _TrvItems_AfterSelect(sender As Object, e As TreeViewEventArgs) Handles _TrvItems.AfterSelect
        If e.Action = TreeViewAction.Unknown Then Return
        ...
End Sub
